jumanji/environments/packing/bin_pack/env.py

```diff
@@ -1017,50 +1153,69 @@ def is_action_allowed(
             item_fits_in_ems = item_fits_in_item(item, item_from_space(ems))
             return ~item_placed & item_mask & ems_mask & item_fits_in_ems
 
-        expanded_obs_state = jax.tree_util.tree_map(
-            functools.partial(jnp.expand_dims, axis=0), obs_ems
-        )
-        expanded_obs_ems_mask = jax.tree_util.tree_map(
-            functools.partial(jnp.expand_dims, axis=0), obs_ems_mask
-        )
-        action_mask = jax.vmap(
-            jax.vmap(is_action_allowed, in_axes=(None, None, 1, 1, 1)),
-            in_axes=(1, 1, None, None, None),
-        )(expanded_obs_state, expanded_obs_ems_mask, items, items_mask, items_placed)
-        action_mask = jnp.moveaxis(action_mask, -1, 0)
-        return jnp.asarray(action_mask, bool)
+        if self.is_rotation_allowed:
+            expanded_obs_state = jax.tree_util.tree_map(
+                functools.partial(jnp.expand_dims, axis=0), obs_ems
+            )
+            expanded_obs_ems_mask = jax.tree_util.tree_map(
+                functools.partial(jnp.expand_dims, axis=0), obs_ems_mask
+            )
+            action_mask = jax.vmap(
+                jax.vmap(is_action_allowed, in_axes=(None, None, 1, 1, 1)),
+                in_axes=(1, 1, None, None, None),
+            )(
+                expanded_obs_state,
+                expanded_obs_ems_mask,
+                items,
+                items_mask,
+                items_placed,
+            )
+            action_mask = jnp.moveaxis(action_mask, -1, 0)
+            return jnp.asarray(action_mask, bool)
+        else:
+            return super()._get_action_mask(
+                obs_ems, obs_ems_mask, items, items_mask, items_placed
+            )
 
     def _ems_are_all_valid(self, state: State) -> chex.Array:
         """Checks if all EMSs are valid, i.e. they don't intersect items and do not stick out of the
         container. This check is only done in debug mode.
         """
         ems_intersection_with_items = jnp.zeros((state.ems_mask.shape), bool)
-
-        for o in range(6):
-            tmp_items = Item(
-                state.items[:][0][o], state.items[:][1][o], state.items[:][2][o]
-            )
-            item_spaces = space_from_item_and_location(tmp_items, state.items_location)
-            ems_intersect_items = jax.vmap(Space.intersect, in_axes=(0, None))(
-                state.ems, item_spaces
+        if self.is_rotation_allowed:
+            for o in range(6):
```

Comment:
I thought at some point you'd vectorised this or was that somewhere elsewhere?

Reply:
I didn't vectorize this, and I didn't think it was worth vectorizing since it is only used in debug mode